src/utils/FixedQueue.ts

```diff
@@ -0,0 +1,55 @@
+import * as _ from 'underscore';
+
+export class FixedQueue<T> {
+
+  private queue: Array<T>;
+  private maxLength: number;
+
+  constructor(initialValues: Array<any> = [], maxLength: number = Number.MAX_VALUE) {
+    this.maxLength = maxLength;
+    this.queue = initialValues;
+    this.trimQueue();
+  }
+
+  getQueue(): Array<T> {
+    return this.queue;
+  }
+
+  push(object: any): FixedQueue<T> {
+    this.queue.push(object);
```

Comment:
I want to return a new instance of Fixed Queue on push to follow the redux immutable state